ykpack/src/types.rs

```diff
@@ -155,38 +155,62 @@ pub enum Operand {
 
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub enum Constant {
-    Scalar(Scalar),
+    UnsignedInt(UnsignedInt),
+    SignedInt(SignedInt),
     Unimplemented,
 }
 
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
-pub enum Scalar {
-    Bits {
-        size: u8,
-        hi_word: u64,
-        lo_word: u64,
-    }, // rmp-serde doesn't support u128.
+pub enum UnsignedInt {
+    Usize(usize),
+    U8(u8),
+    U16(u16),
+    U32(u32),
+    U64(u64),
+    U128{hi: u64, lo: u64},
     Unimplemented,
 }
 
-impl Scalar {
-    pub fn bits_from_u128(size: u8, val: u128) -> Self {
-        Scalar::Bits {
-            size,
-            hi_word: (val >> 64) as u64,
-            lo_word: val as u64,
+impl UnsignedInt {
+    pub fn from_u128(val: u128) -> Self {
```

Comment:
Are we saying that every `UnsignedInt` will be created from a `u128` or ...?

Reply:
No. Just there's a helper for the `u128` (and `i128`) variant since it needs to be split into two for serialisation. Other types are just directly created using the regular constructor.

Why do they need to be split?:
https://github.com/3Hren/msgpack-rust/issues/169

Reply:
OK.